File: easy/1431. Kids With the Greatest Number of Candies/1431. Kids With the Greatest Number of Candies.go
package main

import "slices"

func kidsWithCandies(candies []int, extraCandies int) []bool {
	n := len(candies)
	res := slices.Repeat([]bool{false}, n)
	max_candy := slices.Max(candies)

	for i, candy := range candies {
		res[i] = candy+extraCandies >= max_candy
	}

	return res
}

File: easy/1431. Kids With the Greatest Number of Candies/1431. Kids With the Greatest Number of Candies.py
from typing import List


class Solution:
    def kidsWithCandies(self, candies: List[int], extraCandies: int) -> List[bool]:
        n = len(candies)
        res = [False] * n
        max_candy = max(candies)

        for i in range(len(res)):
            res[i] = True if candies[i] + extraCandies >= max_candy else False

        return res
